lib/media/streaming_engine.js

```diff
@@ -1898,6 +1914,48 @@ shaka.media.StreamingEngine = class {
   }
 
 
+  /**
+   * Fetches and appends a dependency media state
+   *
+   * @param {!shaka.media.StreamingEngine.MediaState_} mediaState
+   * @param {number} presentationTime
+   * @private
+   */
+  async fetchAndAppendDependency_(mediaState, presentationTime) {
+    const dependencyStream = mediaState.stream;
+    const iterator =
+        dependencyStream.segmentIndex.getIteratorForTime(presentationTime);
```

Comment:
Can't you store iterator in media state and keep it with sync with base state? Searching for proper ref may be harmful on some platforms.

Reply:
The problem is that we need the exact time of the base segment, if we re-use the iterator there may be some offset.

Reply:
Something to look into later then.